hal/halx86/smp/mps/mps.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Source File MPS specific functions
+ * COPYRIGHT:   Copyright 2021 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include <hal.h>
+#include <smp.h>
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS ********************************************************************/
+
+PROCESSOR_IDENTITY HalpStaticProcessorIdentity[MAXIMUM_PROCESSORS] = {{0}};
+PPROCESSOR_IDENTITY HalpProcessorIdentity = NULL;
+UINT32 PhysicalProcessorCount = 0;
+
+/* FUNCTIONS ******************************************************************/
+
+VOID
+HalpParseApicTables(IN PLOADER_PARAMETER_BLOCK LoaderBlock)
+{
+    UNIMPLEMENTED;
+}
+
+VOID
+HalpPrintApicTables()
+{ 
+    UINT32 i;
+
+    DPRINT1("HAL has detected a physical processor count of: %d\n", PhysicalProcessorCount);
+
+    for (i = 0; i < PhysicalProcessorCount; i++)
+    {
+        DPRINT1("Information about the following processor is for processors number: %d\n", i);
+        DPRINT1("   The BSPCheck is set to: %X\n", HalpProcessorIdentity[i].BSPCheck);
+        DPRINT1("   The LapicID is set to: %X\n", HalpProcessorIdentity[i].LapicId);
```

Comment:
There's a LAPIC in the SMP MPS ?

Reply:
Yes, the MPS is what we use to find LAPICs before MADT